src/OpenIddict.Models/Application.cs

```diff
@@ -5,16 +5,82 @@
  */
 
 using System;
+using Microsoft.EntityFrameworkCore.Metadata.Internal;
+using System.ComponentModel.DataAnnotations;
 
 namespace OpenIddict.Models {
-    public class Application : Application<string> { }
+    /// <summary>
+    /// The default implementation of <see cref="Application{TKey}"/>
+    ///  which uses a string as a primary key.
+    /// </summary>
+    /// <seealso cref="Application{TKey}" />
+    /// <remarks>
+    /// An an <see cref="P:Application.ApplicationId"/> used <see cref="P:Application.Id"/>.
+    /// So <see cref="P:Application.ApplicationId"/> is not stored in database not to duplicate the same value.
+    /// </remarks>
+    public class Application : Application<string>
+    {
+        public override string ApplicationId
```

Comment:
Looks like rather unusual. Is this a pattern Identity uses somewhere? I'm concerned about introducing something totally new here.